test/ServiceRegistry.js

```diff
@@ -905,6 +911,230 @@ describe("ServiceRegistry", function () {
+                txs[i] = safeContracts.buildSafeTransaction({to: multisig.address, data: callData[i], nonce: 0});
+            }
+            // Remove the original multisig owner and change the threshold
+            // Note that the prevOwner is the very first added address as it corresponds to the reverse order of added addresses
+            // The order in the gnosis safe multisig is as follows: sentinelOwners => agentInstances[last].address => ... =>
+            // => newOwnerAddresses[0].address => serviceOwnerAddress
+            callData.push(multisig.interface.encodeFunctionData("removeOwner", [agentInstances[0].address, serviceOwnerAddress,
+                newMaxThreshold]));
+            txs.push(safeContracts.buildSafeTransaction({to: multisig.address, data: callData[callData.length - 1], nonce: 0}));
+
+            // Build a multisend transaction to be executed by the multisig
+            const safeTx = safeContracts.buildMultiSendSafeTx(multiSend, txs, nonce);
+            const chainId = (await ethers.provider.getNetwork()).chainId;
+            const EIP712_SAFE_TX_TYPE = {
+                // "SafeTx(address to,uint256 value,bytes data,uint8 operation,uint256 safeTxGas,uint256 baseGas,uint256 gasPrice,address gasToken,address refundReceiver,uint256 nonce)"
+                SafeTx: [
+                    { type: "address", name: "to" },
+                    { type: "uint256", name: "value" },
+                    { type: "bytes", name: "data" },
+                    { type: "uint8", name: "operation" },
+                    { type: "uint256", name: "safeTxGas" },
+                    { type: "uint256", name: "baseGas" },
+                    { type: "uint256", name: "gasPrice" },
+                    { type: "address", name: "gasToken" },
+                    { type: "address", name: "refundReceiver" },
+                    { type: "uint256", name: "nonce" },
+                ]
+            };
+
+            // Get the signature of a multisend transaction
+            const signatureBytes = await serviceOwner._signTypedData({verifyingContract: multisig.address, chainId: chainId}, EIP712_SAFE_TX_TYPE, safeTx);
+
+            // Forming Gnosis Safe transaction bytes data: the multisig address itself plus the execTransaction() data
+            // that forms the multisend data with the signature bytes to be executed by the multisig proxy
+            const safeExecData = gnosisSafe.interface.encodeFunctionData("execTransaction", [safeTx.to, safeTx.value,
+                safeTx.data, safeTx.operation, safeTx.safeTxGas, safeTx.baseGas, safeTx.gasPrice, safeTx.gasToken,
+                safeTx.refundReceiver, signatureBytes]);
+
+            // Add the multisig address on top of the multisig exec transaction data
+            const packedData = ethers.utils.solidityPack(["address", "bytes"], [multisig.address, safeExecData]);
```

Comment:
Add a proxy multisig on top of the payload to be decomposed inside the `create()` function